src/widget/notificationedgewidget.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+    Copyright © 2015 by The qTox Project
+
+    This file is part of qTox, a Qt-based graphical interface for Tox.
+
+    qTox is libre software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    qTox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with qTox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "notificationedgewidget.h"
+#include "style.h"
+#include <QLabel>
+#include <QBoxLayout>
+
+#include <QDebug>
+
+NotificationEdgeWidget::NotificationEdgeWidget(Position position, QWidget *parent)
+    : QWidget(parent)
+{
+    setAttribute(Qt::WA_StyledBackground); // Show background.
+    setStyleSheet(Style::getStylesheet(":/ui/notificationEdge/notificationEdge.css"));
+    QHBoxLayout* layout = new QHBoxLayout(this);
+    layout->addStretch();
+
+    QLabel* textLabel = new QLabel(tr("Unread message(s)"), this);
+    textLabel->setMinimumHeight(textLabel->sizeHint().height()); // Prevent cut-off text.
+    layout->addWidget(textLabel);
+
+    QLabel* arrowLabel = new QLabel(this);
+    if (position == Top)
+        arrowLabel->setPixmap(QPixmap("://ui/chatArea/scrollBarUpArrow.svg"));
+    else
+        arrowLabel->setPixmap(QPixmap("://ui/chatArea/scrollBarDownArrow.svg"));
+    layout->addWidget(arrowLabel);
```

Comment:
Just some nitpicking on coding style:
after 
```
if asdf()
    something;
else
    other_thing;
```
there should be an empty newline, before any other code that isn't a part of conditional code.